src/Common/Model/Base.php

```diff
@@ -992,6 +992,67 @@ public function getByIdOrSlug($mIdSlug, $aData = [])
     // --------------------------------------------------------------------------
 
     /**
+     * Fetch an object by its token
+     *
+     * @param  string $sToken The token of the object to fetch
+     * @param  array $aData   Any data to pass to getCountCommon()
+     *
+     * @return \stdClass|null
+     */
+    public function getByToken($sToken, $aData = [])
+    {
+        // @todo if the model does not have a token column, should calls to these methods return empty or throw exception?
```

Comment:
Good shout, something like this ought to do it:

```
if (!$this->tableTokenColumn) {
    throw new ModelException(get_called_class() . '::getByToken() Column variable not set', 1);
}
```

If you could add to all the `getBy*()` methods that'd be just swell :)